testkit/src/network.rs

```diff
@@ -18,9 +18,9 @@ use exonum::{
     blockchain::{ConsensusConfig, ValidatorKeys},
     crypto::{self, PublicKey, SecretKey},
     helpers::{Height, Round, ValidatorId},
-    keys::Keys,
     messages::{Precommit, Propose, Verified},
 };
+use exonum_keys::Keys;
```

Comment:
Can you reexport `exonum_keys` as `exonum::keys` like `merkledb` or `crypto`?

Reply:
To be honest, I don't really like the idea of re-exporting, since it results in a confusing scenarios:

[uno](https://github.com/exonum/exonum/blob/dynamic_services/cli/src/lib.rs#L94) / [dos](https://github.com/exonum/exonum/blob/dynamic_services/examples/cryptocurrency/src/lib.rs#L43) / [tres](https://github.com/exonum/exonum/blob/dynamic_services/exonum/src/blockchain/tests.rs#L26)

The same thing is done *three* different ways. And exonum is already has plenty of such an inconsistencies.

So I don't see any benifits caused by re-exporting, but instead do see only problems.

Can you please explain what's the motivation lies behind the re-exporting policy?

Reply:
All dependencies of Exonum (including MerkleDB and this new crate) are not dependencies of user applications and there is no need in forcing user to handle these additional libraries and their versions in Cargo.toml. Each type which is used in public Exonum API must be available automatically for user. 

Besides clattering Cargo.toml, need in manual dependencies import might also cause interoperability problems between different versions of Exonum and their dependencies.

About your examples - it is definitely a problem, it looks like one of the public reexports of MerkleDB is redundant. Can't say which one exactly - we have problem with `ProtobufConvert` macro when `merkledb` isn't in a scope, I need to check code to give answer. Inconsistency in importing may be solved quite easily - by choosing a method and sticking to it.

> but instead do see only problems.

Can you elaborate if you mean smth other than inconsistency?

Reply:
Yeah, I agree that we should provide users an interface to all the used types.

But for that purpose I'd expect to have a separate module, e.g. `prelude`, which won't only provide re-exports to used crates, but also create a flat import structure (currently users have to build a pretty complex `use` tree just to get things compiled).

And I will be totally fine with such a module.

>Can you elaborate if you mean smth other than inconsistency?

Sort of. Since now we're going to extract several parts of exonum into separate crates, there will be a bigger amount of components (possibly, we wil have more component crates than core modules). And in that scenario, using `exonum::name_of_reexported_crate` inside of the exonum will be sort of confusing -- if you see that, you suppose that it's a module, but actually a separate crate.

As a conclusion:

I'd be happy if we will have a `prelude` module which will export all the required types **for users**, but internally we will use crate names to have an explicit structure of the project.

Reply:
I see. I like your idea about `prelude` module. Yep, I was only talking about users. `use exonum::name_of_reexported_crate` is definitely invalid usage inside Exonum itself, it must be used only in user services (e.g. cryptocurrency).

Reply:
As a conclusion: OK, I'll add a re-export for now, but soon I'll create a PR with `prelude` module. Also I suggest to forbid usage of re-exported crates inside exonum itself.

Is that OK for everybody? cc @aleksuss @alekseysidorov